pkg/sidecar/sidecar.go

```diff
@@ -183,6 +189,101 @@ func (s *Sidecar) dumpBundles(svidResponse *workloadapi.X509Context) error {
 	return nil
 }
 
+func (s *Sidecar) readJSON(fileName string) map[string]interface{} {
+	jsonPath := path.Join(s.config.CertDir, fileName)
+	file, err := os.ReadFile(jsonPath)
+	if err != nil {
+		s.config.Log.Warnf("Unable to read json file: %v", err)
+	}
+
+	certs := make(map[string]interface{})
+	err = json.Unmarshal(file, &certs)
+	if err != nil {
+		s.config.Log.Warnf("Unable to parse json: %v", err)
+	}
+
+	return certs
+}
+
+func (s *Sidecar) writeJSON(fileName string, certs map[string]interface{}) {
+	file, err := json.Marshal(certs)
+	if err != nil {
+		s.config.Log.Warnf("Unable to parse certs: %v", err)
+	}
+
+	jsonPath := path.Join(s.config.CertDir, fileName)
+	err = os.WriteFile(jsonPath, file, os.ModePerm)
+	if err != nil {
+		s.config.Log.Warnf("Unable to write JSON file: %v", err)
+	}
+}
+
+func (s *Sidecar) updateJWTBundle(jwkSet *jwtbundle.Set) {
+	s.config.Log.Info("Updating JWK bundles")
+
+	bundles := make(map[string]string)
+	for _, bundle := range jwkSet.Bundles() {
+		bytes, err := bundle.Marshal()
+		if err != nil {
+			s.config.Log.Warnf("Unable to marshal JWK bundle: %v", err)
+			continue
+		}
+		bundles[bundle.TrustDomain().Name()] = base64.StdEncoding.EncodeToString(bytes)
+	}
+
+	certs := s.readJSON(s.config.JWKFilename)
+	certs["bundles"] = bundles
+	s.writeJSON(s.config.JWKFilename, certs)
+}
+
+func (s *Sidecar) fetchJWTSVID(options ...workloadapi.ClientOption) (*jwtsvid.SVID, error) {
+	clientOptions := workloadapi.WithClientOptions(options...)
+
+	jwtSource, err := workloadapi.NewJWTSource(context.Background(), clientOptions)
+	if err != nil {
+		s.config.Log.Warnf("Unable to create JWTSource: %v", err)
+		return nil, err
+	}
+	defer jwtSource.Close()
+
+	jwtSVID, err := jwtSource.FetchJWTSVID(context.Background(), jwtsvid.Params{Audience: s.config.JWTAudience})
+	if err != nil {
+		s.config.Log.Warnf("Unable to fetch JWT SVID: %v", err)
+		return nil, err
+	}
+
+	_, err = jwtsvid.ParseAndValidate(jwtSVID.Marshal(), jwtSource, []string{s.config.JWTAudience})
+	if err != nil {
+		s.config.Log.Warnf("Unable to parse or validate token: %v", err)
+		return nil, err
+	}
+
+	return jwtSVID, nil
+}
+
+func (s *Sidecar) updateJWTSVID(ctx context.Context, options ...workloadapi.ClientOption) {
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		default:
+			s.config.Log.Infof("Updating JWT SVID")
+			jwtSVID, err := s.fetchJWTSVID(options...)
+			if err != nil {
+				time.Sleep(time.Second)
+				continue
+			}
+
+			certs := s.readJSON(s.config.JWTFilename)
+			certs["svid"] = jwtSVID.Marshal()
```

Comment:
Why are we parsing the svid element out? I thought this file would contain only the token?

Reply:
Well, the origin of the PR is the [Issue #43](https://github.com/spiffe/spiffe-helper/issues/43), maybe this file could only contain the JWT SVID. What do you think @faisal-memon ?